packages/dhis2/src/Utils.js

```diff
@@ -66,8 +80,51 @@ export function generateUrl(configuration, options, resourceType, path = null) {
 
   console.log(apiMessage);
 
-  const url = buildUrl(urlString, hostUrl, apiVersion);
+  const urlPath = buildUrl(urlString, apiVersion);
+  if (path) return `${urlPath}/${path}`;
+  return urlPath;
+}
+export const configureAuth = (auth, headers = {}) => {
+  if ('pat' in auth) {
+    Object.assign(headers, {
+      Authorization: `ApiToken ${auth.pat}`,
+    });
+  } else if ('password' in auth) {
+    Object.assign(headers, makeBasicAuthHeader(auth.username, auth.password));
+  } else {
+    throw new Error(
+      'Invalid authorization credentials. Include an pat, username or password in state.configuration'
+    );
+  }
+
+  return headers;
+};
+
+export async function request(configuration, requestData) {
+  const { hostUrl } = configuration;
+  const { method, path, options = {}, data = {} } = requestData;
+
+  const {
+    headers = { 'content-type': 'application/json' },
+    query = {},
+    parseAs = 'json',
+  } = options;
+
+  console.log(`Sending ${method} request to ${path}`);
```

Comment:
Do we want this log? because common HTTP should log the response for us (whether it failed or not). So this feels like spam.